### PR TITLE
Fix npm issues

### PR DESCRIPTION
**Pull request by briandrawert**

### CODE REVIEW

**Review by BryanRumsey — CHANGES_REQUESTED** [review]
Tested on MacOS and the npm changes work with `make`, `make watch`, `make hub`, and `make run_hub_staging`.  The launch web browser script doesn't work.

**Review by jtcooper10 — CHANGES_REQUESTED** [review]
Does not work on Windows, but the fixes should be very simple (comments included). Getting the same `ModuleNotFound` error as Bryan, as well.

Other than that, everything works great. Tested on Windows 11.

File: launch_webbrowser.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env python3
+import sys
+import os
+import docker
```
[the remark this answers is not in the document]
Comment (BryanRumsey) [reply launch_webbrowser.py:4]: installing docker.py fixed this issue

**Review by BryanRumsey — APPROVED** [review]

**Review by seanebum — APPROVED** [review]

File: Makefile [hunk 2]
```diff
@@ -127,29 +127,40 @@ test:   create_working_dir
 build_and_test: build test
 
 run:    create_working_dir
+	python3 launch_webbrowser.py &
```
[the remark this answers is not in the document]
Comment (briandrawert) [reply Makefile]: Should be resolved, please re-check.

**Review by jtcooper10 — APPROVED** [review]

**Review by BryanRumsey — APPROVED** [review]

File: launch_webbrowser.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        url_regex = r"(?i)\b((?:https?://|www\d{0,3}[.]|[a-z0-9.\-]+[.][a-z]{2,4}/)(?:[^\s()<>]+|\(([^\s()<>]+|(\([^\s()<>]+\)))*\))+(?:\(([^\s()<>]+|(\([^\s()<>]+\)))*\)|[^\s`!()\[\]{};:'\".,<>?«»“”‘’]))"
+        jupyter_url_bytes = jupyter_url_generator.output
+        jupyter_url_string = jupyter_url_bytes.decode("utf-8")
+        jupyter_url_list = re.findall(url_regex,jupyter_url_string)
+        if len(jupyter_url_list) == 0: continue
+        jupyter_url_sequence = jupyter_url_list[0]
+        jupyter_url=jupyter_url_sequence[0]
+        jupyter_url=jupyter_url.replace('0.0.0.0','127.0.0.1')
+        print(f"Opening StochSS webpage...\n")
+        webbrowser.open_new(jupyter_url)
```
Comment (seanebum) [inline launch_webbrowser.py:43]: `webbrowser.open_new_tab(jupyter_url)` may be preferrable here

**Review by seanebum — APPROVED** [review]

**Pull request merged**